src/components/AnonymousChat/PostNotificationChannelItem.tsx

```diff
@@ -109,13 +97,16 @@ const AnonPostNotificationChannelItem: (props: MessageChannelItemProps) => React
       postNotificationMessageText={postNotificationMessageText}
       postNotificationMessageUser={commenterName}
       postNotificationPicture={postNotificationPicture}
-      showPostNotificationStats={isOwnPost}
       time={calculateTime(item?.lastUpdatedAt, true)}
-      type={type}
       unreadCount={postNotifItem?.unreadCount}
       upvote={item?.rawJson?.upvote}
+      channelType={item?.channelType}
     />
   );
 };
 
-export default AnonPostNotificationChannelItem;
+const isEqual = (prevProps, nextProps) => {
+  return nextProps.item === prevProps.item;
+};
+
+export default React.memo(PostNotificationChannelItem, isEqual);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The component is now wrapped in `React.memo` with a custom comparison function `isEqual`. This can improve performance by preventing unnecessary re-renders, but make sure that the comparison function correctly determines when the props have changed. The removed prop `showPostNotificationStats` and added prop `channelType` should also be properly handled in the `BaseChannelItem` component.

```diff
- showPostNotificationStats={isOwnPost}
+ channelType={item?.channelType}
- export default AnonPostNotificationChannelItem;
+ const isEqual = (prevProps, nextProps) => {
+   return nextProps.item === prevProps.item;
+ };
+ 
+ export default React.memo(PostNotificationChannelItem, isEqual);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->